app/models/user_role.rb

```diff
@@ -49,4 +57,17 @@ def self.grant(user, role, facility = nil)
 
   validates_presence_of :user_id
   validates_inclusion_of :role, in: valid_roles, message: "is not a valid value"
+  validates_uniqueness_of :role,
+    scope: [:facility_id, :user_id],
+    message: I18n.t("activerecord.errors.models.user_role.duplicate")
+  validates_with UserRoleFacilityValidator
+
+  def facility_role?
+    self.class.facility_roles.include?(role)
+  end
+
+  def global_role?
+    !facility_role?
```

Comment:
(somewhat related to my last comment above), it seems like we should use the same logic for `global_role?` as the scope `global`